packages/shared/lib/utils.ts

```diff
@@ -47,77 +45,91 @@ export const generateRandomId = (): string => {
+            }
+        } else {
+            addError({ time: Date.now(), type: "deepLink", message: `Error handling deep link. Does not start with iota://` })
+        }
+    } catch (err) {
+        addError({ time: Date.now(), type: "deepLink", message: `Error handling deep link. ${err.message}` })
     }
 }
 
-/** Parse an IOTA address input
- * @param {string} input
- * @returns {ParsedURL} - The parsed address, message and/or amount values
+/**
+ * Parse a deep link for the wallet (iota://wallet/send)
+ * @param  {url} The url
+ * @param  {pathParts} The path parts
+ * @return {ParsedURL}  The parsed address, message and/or amount values
  */
-export const parseAddress = (input) => {
-    const result = {
-        address: null,
-        message: null,
-        amount: null,
+export const parseWalletSendDeepLink = (addressPrefix, url, pathParts) => {
+    if (pathParts.length === 0) {
+        addError({ time: Date.now(), type: "deepLink", message: `No address part in the url path` })
+        return
     }
 
-    if (!input || typeof input !== 'string') {
-        return null
-    }
+    // Path starts with '/' so ignore the first one
+    const address = pathParts[0]
 
-    if (input.match(VALID_MAINNET_ADDRESS) || input.match(VALID_DEVNET_ADDRESS)) {
-        result.address = input
-        return result
+    if (!new RegExp(`^${addressPrefix}1[02-9ac-hj-np-z]{59}$`).test(address)) {
+        addError({ time: Date.now(), type: "deepLink", message: `Address '${address}' does not match prefix '${addressPrefix}' or format` })
+        return
     }
 
-    try {
-        let parsed = {
-            address: null,
-            message: null,
-            amount: null,
-        }
-
-        if (input.toLowerCase().indexOf('iota:') === 0) {
-            const url = new URL(input)
-            parsed.address = url.hostname.toLowerCase()
-            parsed.message = url.searchParams.get('message')
-            parsed.amount = url.searchParams.get('amount')
-        } else {
-            parsed = JSON.parse(input)
-        }
+    const amountParam = url.searchParams.get('amt')
+    const parsedAmount = Number.parseFloat(amountParam)
+    if (Number.isNaN(parsedAmount) || !Number.isFinite(parsedAmount)) {
+        addError({ time: Date.now(), type: "deepLink", message: `Amount is not a number '${amountParam}'` })
+        return
+    }
 
-        if (parsed.address.match(VALID_MAINNET_ADDRESS) || parsed.address.match(VALID_DEVNET_ADDRESS)) {
-            result.address = parsed.address
-        } else {
-            return null
-        }
+    const unitParam: Unit = url.searchParams.get('unit') as Unit ?? Unit.i
+    if (!Object.values(Unit).includes(unitParam)) {
+        addError({ time: Date.now(), type: "deepLink", message: `Unit is not recognised '${unitParam}'` })
+    }
 
-        if (parsed.message && typeof parsed.message === 'string') {
```

Comment:
wouldn't there be maxLength for the message we might want to validate?

Reply:
ok just to add to this. I'm not expert on this but do we have to worry about XSS attack?